src/braket/aws/aws_session.py

```diff
@@ -825,3 +825,27 @@ def copy_session(
         # Preserve user_agent information
         copied_session._braket_user_agents = self._braket_user_agents
         return copied_session
+
+    def get_full_image_tag(self, image_uri: str) -> str:
```

Comment:
Can this be cached since it is an expensive operation and is unlikely to change within a user's session?

Reply:
Good idea!